frontend/src/components/Conversation/Conversation.tsx

```diff
@@ -51,18 +60,14 @@ export const Conversation = () => {
         // empty array and the mutation's onSuccess properly populates the new messages + results
         { ...result, result_id: generateUUID() },
       ]),
-    onSettled: () => setStreamedResults([]),
+    onSettled: (_, error) => {
+      setStreamedResults([]);
+      if (!error && messages && messages.length < 2) {
```

Comment:
So this will be called twice when the first message is returned and then when the second message is returned?

Reply:
Okay, yeah would be nice to add a comment on why 2.